apps/api/src/controllers/leaderboards-controller.ts

```diff
@@ -0,0 +1,140 @@
+import { logger } from '@codeheroes/common';
+import { TimeBasedActivityStats, Collections } from '@codeheroes/types';
+import * as express from 'express';
+import { getFirestore } from 'firebase-admin/firestore';
+
+const router = express.Router();
+
+interface TimePeriod {
```

Comment:
Instead of a modular approach, I'm bringing all this stuff (db connection, firestore query, interfaces) directly to the controller. Reason: because this is very much experimental, just to get some idea how the leaderboards could finally work.